app/controllers/Application.java

```diff
@@ -165,7 +181,23 @@ public static Result search() {
           .where()
           .idEq(appId).findUnique();
       return ok(searchPage.render(null, jobDetails.render(result)));
-    } else if (flowExecId != null && !flowExecId.isEmpty()) {
+    } else if (Utils.isSet(flowExecId)) {
+      String bestFlowExecId;
+      // check for exact match
+      bestFlowExecId = bestMatchForFlowExecIdLikeValue(flowExecId);
```

Comment:
Performance seems fine for us. Full text search will only have to happen on prefix matching. Suffix matches can still use the index and we try to do that before we try to prefix match. If you are concerned we could consider eliminating the prefix matching step.